Create a new xmode "Docs"

Carreau opened the pull request:
This is a compact mode with a minimal header and no traceback,
mosttly to be used in Documentation, where the context of the traceback
is rarely useful

```
$ ipython --InteractiveShell.xmode=Docs

In [1]: import pywt
    ...:
    ...: a = pywt.idwt([1,2,3,4,5], [1,2,3,4], 'db2' ,'symmetric')
    ...:
    ...: print(a)
Traceback (most recent call last): Cell In[1], line 3

->  a = pywt.idwt([1,2,3,4,5], [1,2,3,4], 'db2' ,'symmetric')

    [... 3 skipped frames]

ValueError: Coefficients arrays must have the same size.
```

Carreau committed:
Some ruff rule enabling and fixing.

Carreau committed:
test

Carreau committed:
fix more

Carreau commented:
This should take care of #14591 

@rgommers, tell me if you find that the following is compact enough, or if you want something even more compact.

```
Traceback (most recent call last): Cell In[1], line 3

->  a = pywt.idwt([1,2,3,4,5], [1,2,3,4], 'db2' ,'symmetric')

    [... 3 skipped frames]

ValueError: Coefficients arrays must have the same size.
```

rgommers commented:
That looks good to me, I think that should be enough. Thanks for pushing on this!

Carreau commented:
Thanks, I still need to clean that up, might a few more days though.

Thanks for the quick feedback.

Carreau committed:
Create a new xmode "Docs"

This is a compact mode with a minimal header and no traceback,
mosttly to be used in Documentation, where the context of the traceback
is rarely useful

```
$ ipython --InteractiveShell.xmode=Docs

In [1]: import pywt
   ...:
   ...: a = pywt.idwt([1,2,3,4,5], [1,2,3,4], 'db2' ,'symmetric')
   ...:
   ...: print(a)
Traceback (most recent call last): Cell In[1], line 3

->  a = pywt.idwt([1,2,3,4,5], [1,2,3,4], 'db2' ,'symmetric')

    [... 3 skipped frames]

ValueError: Coefficients arrays must have the same size.
```

Carreau committed:
properly call self

Carreau committed:
reformat/isort

Carreau committed:
reformat before vlack upgrade

Carreau committed:
reformat after black upgrade
